src/frontend/src/handler/drop_mv.rs

```diff
@@ -68,13 +70,27 @@ pub async fn handle_drop_mv(
             _ => return Err(table.bad_drop_error()),
         }
 
-        table.id()
+        (table.id(), table.stream_job_status)
     };
 
-    let catalog_writer = session.catalog_writer()?;
-    catalog_writer
-        .drop_materialized_view(table_id, cascade)
-        .await?;
+    match status {
+        StreamJobStatus::Created => {
+            let catalog_writer = session.catalog_writer()?;
+            catalog_writer
+                .drop_materialized_view(table_id, cascade)
+                .await?;
+        }
+        StreamJobStatus::Creating => {
```

Comment:
nit: We currently ignore `cascade` here while it could be meaningful if we're going to impl https://github.com/risingwavelabs/risingwave/issues/12771.

Reply:
Yeah agree.

Reply:
Maybe we can use `drop_materialized_view` directly and let meta handle whether to `drop` or `cancel` this mv. The benifit is that we can reuse the version field of drop RPC without adding a new one for cancel.